core/sql/common/CharType.cpp

```diff
@@ -1217,7 +1217,7 @@ void CharType::minMaxRepresentableValue(void* bufPtr,
   if (stringLiteral)
     {
       NABoolean isNull = FALSE;
-      NABoolean res = createSQLLiteral((const char *) bufPtr, *stringLiteral, isNull, h);
+      NABoolean res = createSQLLiteral((const char *) bufPtr  - getSQLnullHdrSize(), *stringLiteral, isNull, h);
```

Comment:
Could you please explain this line.
Is the understanding that bufPtr passed in does not include null header, but the previous 2 bytes (say type is nullable and null size is 2 bytes) are available and meant to hold null bytes. I suppose I am nervous about taking a pinter passed in and going a few bytes before. How do we know that is still a valid location? Maybe all this handled in createSQLLiteral ?